### PR TITLE
Develop 159 remaining mpi

### LINKED ISSUE #159: Add MPI pack size, pack, and unpack functions to remaining classes

### PR DESCRIPTION
**Pull request by mattldawson**
closes #159

also includes an update to the `musica-core` library that makes the MPI communicator a required input to MPI functions

### CODE REVIEW

File: docs/source/developer/science_contributions.rst [hunk 1]
```diff
@@ -1025,6 +1065,151 @@ It will also serve as an example for how users can configure and use your new su
...
+members of the base class from the character buffer, and is
+required.
+Similar to the ``pack_size( )`` function, this subroutine makes use of
+the generic ``musica_mpi_unpack( )`` function for unpacking primitive
+Fortran data types from a character buffer (see the
+`musica core <https://ncar.github.io/musica-core/html/namespacemusica__mpi.html>`_
+library documentation for more details).
+The ``assert( )`` call helps with debugging MPI errors and ensures
+that the data you packed fits in the pack size from the ``pack_size( )``
+function.
```
Comment (K20shores) [inline docs/source/developer/science_contributions.rst:1211]: I think we should add a recommendation to included tests for anything that requires MPI additions

Comment (mattldawson) [reply docs/source/developer/science_contributions.rst:1211]: I agree - will add one

File: docs/source/developer/science_contributions.rst [hunk 2]
```diff
@@ -1073,7 +1258,91 @@ The `musica_assert <https://ncar.github.io/musica-core/html/namespacemusica__ass
 module contains a number of functions that can be useful in
 unit tests.
 
-You will need to modify the ``CMakeLists.txt`` file in the
+If your new class requires the MPI functions ``pack_size( )``, ``mpi_pack( )``,
```
Comment (K20shores) [inline docs/source/developer/science_contributions.rst:1261]: Can you also add instructions on running the MPI tests here?

Comment (mattldawson) [reply docs/source/developer/science_contributions.rst:1261]: they should automatically run if MPI support is compiled in. Although, I don't think we have instructions on how to build with MPI support yet. Should we add this to the Installation section?

Comment (mattldawson) [reply docs/source/developer/science_contributions.rst:1268]: I added some text to the Installation instructions and to the Test creation section about building/testing with MPI

**Pull request merged**